test/RewardsStreamerMP.t.sol

```diff
@@ -2932,62 +3004,85 @@ contract FuzzTests is RewardsStreamerMPTest {
-                rewardIndex: 0
-            })
-        );
+        if (expectedAliceParams.vaultBalance > 0) {
+            uint256 rawAccruedMP = _accrueMP(stakeAmount, accruedTime);
+            expectedAliceParams.mpAccrued =
+                Math.min(expectedAliceParams.mpAccrued + rawAccruedMP, expectedAliceParams.maxMP);
+            expectedSystemState.totalMPAccrued =
+                Math.min(expectedSystemState.totalMPAccrued + rawAccruedMP, expectedSystemState.totalMaxMP);
+        }
 
-        checkVault(
-            CheckVaultParams({
-                account: vaults[alice],
-                rewardBalance: 0,
-                stakedBalance: stakeAmount,
-                vaultBalance: stakeAmount,
-                rewardIndex: 0,
-                mpStaked: expectedStakedMP,
-                mpAccrued: expectedTotalMP,
-                maxMP: expectedMaxTotalMP,
-                rewardsAccrued: 0
-            })
-        );
+        vm.warp(vm.getBlockTimestamp() + accruedTime);
+
+        streamer.updateGlobalState();
+        streamer.updateVaultMP(vaults[alice]);
+
+        check("Accrue: ", expectedSystemState);
+        check("Accrue: ", expectedAliceParams);
+
+        if (lockUpPeriod2 == 0) {
+            expectedRevert = IStakeManager.StakingManager__DurationCannotBeZero.selector;
+        } else {
+            if (expectedAliceParams.vaultBalance == 0) {
+                expectedRevert = StakeMath.StakeMath__InsufficientBalance.selector;
+            } else if (lockUpPeriod2 > MAX_LOCKUP_PERIOD) {
+                expectedRevert = StakeMath.StakeMath__InvalidLockingPeriod.selector;
+            } else {
+                uint256 dt_lock =
+                    (accruedTime <= lockUpPeriod) ? (lockUpPeriod - accruedTime) + lockUpPeriod2 : lockUpPeriod2;
+                if (dt_lock >= MIN_LOCKUP_PERIOD && dt_lock <= MAX_LOCKUP_PERIOD) {
+                    if (lockUpPeriod + lockUpPeriod2 > MAX_LOCKUP_PERIOD) {
+                        //expectedRevert = FuzzTests.FuzzTests__UndefinedError.selector;
+                        expectedRevert = StakeMath.StakeMath__AbsoluteMaxMPOverflow.selector;
+                    } else {
+                        expectedRevert = NO_REVERT;
+                        uint256 additionalBonusMP = _bonusMP(stakeAmount, lockUpPeriod2);
+                        expectedSystemState.totalMPStaked += additionalBonusMP;
+                        expectedSystemState.totalMPAccrued += additionalBonusMP;
+                        expectedSystemState.totalMaxMP += additionalBonusMP;
+                        expectedAliceParams.mpStaked += additionalBonusMP;
+                        expectedAliceParams.mpAccrued += additionalBonusMP;
+                        expectedAliceParams.maxMP += additionalBonusMP;
+                    }
+                } else {
+                    expectedRevert = StakeMath.StakeMath__InvalidLockingPeriod.selector;
+                }
+            }
+        }
```

Comment:
Blocks like these are somewhat hard to parse.

How do you feel about introducing fuzz tests for the revert cases only, instead of having one fuzz test trying to capture all the revert cases as well (next to the other non-reverting cases).

That way we can group the tests nice (non-reverting and reverting) and it's easier to follow what's happening.